src/service/custom-element-registry.js

```diff
@@ -135,6 +135,9 @@ export function stubElementsForDoc(ampdoc) {
     ampdoc.declareExtension(extensionId, extensionVersion);
     stubElementIfNotKnown(ampdoc.win, extensionId);
   });
+  if (ampdoc.isBodyAvailable()) {
+    ampdoc.setExtensionsKnown();
+  }
```

Comment:
Don't we need a similar set for single doc?

Reply:
This is actually the single doc mode. The multidoc/shadowdoc is supported in the `multidoc-manager.js` and FIE is in `friendly-iframe-embed.js`.

Reply:
Checked, and there's a second call in `src/runtime.js` that does just that.